BoxBrowseSDK/BoxBrowseSDK/Categories/UIImage+BOXBrowseSDKAdditions.h

```diff
@@ -59,6 +59,20 @@ NS_ASSUME_NONNULL_BEGIN
  */
 + (UIImage *)box_genericFolderIcon;
 
+/**
```

Comment:
Is there a reason we wouldn't just use '+ (UIImage *)box_iconForItem:(BOXItem *)item' or '+ (UIImage *)box_smallIconForItem:(BOXItem *)item' for the use cases that we're introducing this for? The reason I bring this up is that we want to avoid the creation of duplicated logic in all kinds of places to figure out what kind of icon to use for different items. Generic icons are exposed for general purpose use, but otherwise we should be able to know the item and have this SDK just give us the right icon to use for it

Reply:
There's no BoxItem to leverage this case.

Imagine someone wanting to build a UI (using our SDKs) and in that UI they wanted to use a collab-folder icon, independent of any *specific* collaborated folder, they ought to be able to fetch that specific icon.

Further, we have a public API for getting a file icon independent of a BoxItem (all you need is a filename String). So why can't one also get the various folder icons independent of a BoxItem.

Reply:
I already covered the why above.
The more relevant conversation is having a legitimate use-case for it. I think that case is a collaboration notification, so you may not have access to the item until you accept (or satisfy some security pre-conditions).
In that case, we do need to expose this kind of thing. We should, however, use it in box_iconForItem so we still have a centralized location to define what the icon associated with this type of folder is.

Also, for your use-case do we not need the "smallIcon"?